### PR TITLE
Fixes #27 #77 Improved functions for add, edit and search book

### LINKED ISSUE #27: Code Cleanup
This issue is created to track any code cleanup which is present in the codebase and would be required. 

This is not an urgent task atm. But Needs to be taken care when the main functionality of the application starts working. 

Keep this issue open till all codebase is cleared. Multiple PRs can be referenced to this issue. 

Thanks!

### PR DESCRIPTION
**Pull request by shriaas2898**
There were a lot of problems in making tables responsive specially if the contain a form, removed `listbook` edit page which contained huge table of book info.
* Wrote a function add  which adds a book in the DB and sets appropriate message in `$_SESSION["success"]`/`$_SESSION["error"]`. This function does not contain HTML code and is called when the form is submitted.
* Wrote a function edit which updates book information in DB  and sets appropriate message in `$_SESSION["success"]`/`$_SESSION["error"]`
     - Also added code to update image using file picker instead of specifying the path.
     - When the cover image is updated in the database the corresponding old file is deleted from files.

* Wrote a function `delete` which deletes the book record from the table and its cover image from files.
* Wrote a function `search` which searches book by `title`, `author` or `category`.
* Wrote a function `searchById` which searches book by its `id`.
* Added an option to edit When admin searches any book.
* Reused `AddBook` page to edit book when id is passed as a `GET` parameter.
## Previous edit book flow
Go to list book edit and update details
image

## Current edit book flow:
1. Search book -> edit
image
2. Edit details
image

This a huge commit would recommend everyone to review it. @JayKandari @rahulsavaria @guptahemant @AJV009 @anjali-rathod @pragzii3896

### CODE REVIEW

**Comment by AJV009** [comment]
Nice 👍👍👍
I'll soon make a pull from your branch and test em! 😅

File: public-html/addbook.php [hunk 1]
```diff
@@ -24,15 +45,101 @@
...
 	$menu->render_menu();
-	echo '<div class="main_content">';
-	$k = new book;
-	$row = $k->add();
-	echo '</div">';
-	?>
+	
+	?> 
+	<div class="main_content">
+	<div class="container">
```
Comment (rahulsavaria) [inline public-html/addbook.php]: incorrect indentation

**Review by rahulsavaria — CHANGES_REQUESTED** [review]

**Comment by shriaas2898** [comment]
@rahulsavaria made the requested changes

File: public-html/addbook.php [hunk 2]
```diff
@@ -7,31 +7,136 @@
...
+      if ($book_details) {
+            // If book_id is supplied through URL populate the form
+            $title = $book_details['title'];
+            $category = $book_details['category'];
+            $author = $book_details['author'];
+            $added_on = $book_details['added_on'];
+            $qty = $book_details['qty'];
+            $bcover = $book_details['ipath'];
+      } else {
+            $_SESSION["error"] = "No books found";
```
Comment (guptahemant) [inline public-html/addbook.php:30]: Please add a follow up to replace to use of session variable directly with a get and set method in session class. Also if possible add appropriate methods to store and show session messages from session variables.

**Comment by AJV009** [comment]
@shriaas2898 
The edituser table seems to be messed up
check this out -> https://vod-progressive.akamaized.net/exp=1602107008~acl=%2A%2F2067490946.mp4%2A~hmac=2df89f68267d83280e22ad547e893b767986fc637bc25e7e70bdbd20ab2e9d97/vimeo-prod-skyfire-std-us/01/3184/18/465921565/2067490946.mp4

File: public-html/searchbook.php [hunk 3]
```diff
@@ -7,7 +7,8 @@
 
 use src\SessionCookie;
 use template\Menu;
-use src\Connection;
+// use src\Connection;
```
Comment (guptahemant) [inline public-html/searchbook.php]: remove this if not needed.

File: public-html/searchbook.php [hunk 4]
```diff
@@ -74,52 +75,47 @@
...
                      <td data-label='Author'>" . $row['author'] . "</td>
                      <td data-label='Category'>" . $row['category'] . "</td><td>";
-                                if (isset($_SESSION["logged_in"])) {
-                                    echo '<form method="post">';
-                                    echo '<input type="hidden" name="bid" value="' . $row["id"] . '"/>';
-                                    echo '<button name="issue">ISSUE BOOK </button>';
-                                    echo '</form>';
-                                } else {
-                                    echo '<h4><a href="login.php"><button>Login to issue book</button></a></h4>';
+                            if (isset($_SESSION["logged_in"])) {
```
Comment (guptahemant) [inline public-html/searchbook.php]: use the session class to detect if a user is logged in.

File: resources/src/books.php [hunk 5]
```diff
@@ -41,119 +42,138 @@ public function disp_book($recent)
...
+    * Inserts book details in `book` table, and sets `$_SESSION["success"]` on successful execution 
+    * else sets `$_SESSION["error"]` if some error is encountered.
+    * @param array $book_details  
+    * Associative array which contains details of book to be added in this order:
+    *  ["title", "author","category","added_on", "qty","available","bcover"]
+    * @return void 
+    */
+   public function add($book_details)
+   {
+      var_dump($book_details);
```
Comment (guptahemant) [inline resources/src/books.php]: Remove debug statement.

**Review by guptahemant — CHANGES_REQUESTED** [review]

**Comment by shriaas2898** [comment]
> @shriaas2898
> The edituser table seems to be messed up
> check this out -> https://vod-progressive.akamaized.net/exp=1602107008~acl=%2A%2F2067490946.mp4%2A~hmac=2df89f68267d83280e22ad547e893b767986fc637bc25e7e70bdbd20ab2e9d97/vimeo-prod-skyfire-std-us/01/3184/18/465921565/2067490946.mp4

This says access denied. Also edit user table's UI is messed up and I haven't worked on it we have to remove tables anyways.

**Comment by AJV009** [comment]
> > @shriaas2898
> > The edituser table seems to be messed up
> > check this out -> https://vod-progressive.akamaized.net/exp=1602107008~acl=%2A%2F2067490946.mp4%2A~hmac=2df89f68267d83280e22ad547e893b767986fc637bc25e7e70bdbd20ab2e9d97/vimeo-prod-skyfire-std-us/01/3184/18/465921565/2067490946.mp4
> 
> This says access denied. Also edit user table's UI is messed up and I haven't worked on it we have to remove tables anyways.

:see_no_evil:  :laughing: 
I thought other tables would get effected too

File: public-html/addbook.php (hunk 2 above)
Comment (shriaas2898) [reply public-html/addbook.php:30]: Sure I'll create an issue for this.

**Review by guptahemant — APPROVED** [review]

**Review by rahulsavaria — APPROVED** [review]

**Pull request merged**